Examples/Algorithms/TrackFindingExaTrkX/src/PrototracksToParsAndSeeds.cpp

```diff
@@ -0,0 +1,201 @@
+    // case, we want to keep the whole prototrack. Therefore, we operate on a
+    // tmpTrack.
+    std::sort(track.begin(), track.end(), [&](auto a, auto b) {
+      if (indexToGeoId[a].volume() != indexToGeoId[b].volume()) {
+        return indexToGeoId[a].volume() < indexToGeoId[b].volume();
+      }
+      return indexToGeoId[a].layer() < indexToGeoId[b].layer();
+    });
+
+    tmpTrack.clear();
+    std::unique_copy(
+        track.begin(), track.end(), std::back_inserter(tmpTrack),
+        [&](auto a, auto b) {
+          return indexToGeoId[a].volume() == indexToGeoId[b].volume() &&
+                 indexToGeoId[a].layer() == indexToGeoId[b].layer();
+        });
+
+    // in this case we cannot seed properly
+    if (tmpTrack.size() < 3) {
+      ACTS_DEBUG(
+          "Cannot seed because less then three hits with unique (layer, "
+          "volume)");
+      skippedTracks++;
+      continue;
+    }
+
+    // Make the seed
+    tmpSps.clear();
+    std::transform(track.begin(), track.end(), std::back_inserter(tmpSps),
+                   [&](auto i) { return indexToSpacepoint[i]; });
+    tmpSps.erase(std::remove_if(tmpSps.begin(), tmpSps.end(),
+                                [](auto sp) { return sp == nullptr; }),
+                 tmpSps.end());
+
+    if (tmpSps.size() < 3) {
+      ACTS_WARNING("Could not find all spacepoints, skip");
+      skippedTracks++;
+      continue;
+    }
+
+    std::sort(tmpSps.begin(), tmpSps.end(),
+              [](const auto &a, const auto &b) { return a->r() < b->r(); });
+
+    // Simply use r = m*z + t and solve for r=0 to find z vertex position...
+    // Probably not the textbook way to do
+    const auto m = (tmpSps.back()->r() - tmpSps.front()->r()) /
+                   (tmpSps.back()->z() - tmpSps.front()->z());
+    const auto t = tmpSps.front()->r() - m * tmpSps.front()->z();
+    const auto z_vertex = -t / m;
+    const auto s = tmpSps.size();
+
+    SimSeed seed =
+        m_cfg.buildTightSeeds
+            ? SimSeed(*tmpSps[0], *tmpSps[1], *tmpSps[2], z_vertex)
+            : SimSeed(*tmpSps[0], *tmpSps[s / 2], *tmpSps[s - 1], z_vertex);
+
+    // Compute parameters
+    const auto geoId = seed.sp()
+                           .front()
+                           ->sourceLinks()
+                           .front()
+                           .template get<IndexSourceLink>()
+                           .geometryId();
+    const auto &surface = *m_cfg.geometry->findSurface(geoId);
+
+    auto pars = Acts::estimateTrackParamsFromSeed(
+        {}, seed.sp().begin(), seed.sp().end(), surface, {0., 0., 2_T}, 0.0);
+
+    if (not pars) {
+      ACTS_WARNING("Skip track because of bad params");
+    }
+
+    seededTracks.push_back(track);
+    seeds.emplace_back(std::move(seed));
+    parameters.push_back(
+        Acts::BoundTrackParameters(surface.getSharedPtr(), *pars, m_covariance,
+                                   Acts::ParticleHypothesis::pion()));
```

Comment:
what is the difference to truth estimated seeding?

to me it looks like it is petty much doing the same but also giving you an aligned prototrack vector

do you see a way to unify those two?

Reply:
This is truth estimated seeding.

The problem is, I cannot do:
```
prototracks --|PrototracksToSeeds|--> seeds --|TrackParamEstimation|--> params
```
because if a seed fails in `TrackParamEstimation`, there is no way to get a well behaved prototrack collection.